students/charlie_england/lesson02/assignment/src/charges_calc.py

```diff
@@ -0,0 +1,138 @@
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+import logging
+
+def parse_cmd_arguments():
+    """
+    creates an argument parser, adds 3 arguments
+    i for input, o for output and optional d for debugging with 3 levels
+    """
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='level of debug, 0 is default = not set, 1 = critical, 2 = warning, 3 = debug', required=False)
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    Opens JSON file and returns data,
+    logs a critical error if there is an error
+    """
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+            #Json decode error, expecting property name enclosed in double quotes
+        except Exception as exception:
```

Comment:
This exception will catch all issues but you should try to be more exact about the errors you catch so that the user is aware what went wrong. As your comment above, one possible issue is the JsonDecodeError. Another one to consider is a missing file error.